configure.ac

```diff
@@ -75,15 +79,48 @@ AC_SEARCH_LIBS([lrint], [m], [
 ])
 pkg_libs="$LIBS"
 
+AS_IF([test "x$enable_pthreads" = xyes && test "x$enable_w32threads" = xyes], [
+    AC_MSG_ERROR([both pthreads and w32threads were requested.])
+])
+
+AS_IF([(test "x$enable_pthreads" = xyes || test "x$enable_w32threads" = xyes) && test "x$ac_cv_header_stdatomic_h" != xyes], [
+    AC_MSG_ERROR([thread support was requested, but <stdatomic.h> was not found.])
+])
+
+pthread_found=false
+
+AS_IF([test "x$enable_pthreads" != xno && test "x$ac_cv_header_stdatomic_h" = xyes], [
+    AX_PTHREAD([
+        pkg_libs="$pkg_libs $PTHREAD_LIBS"
+        CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
+        LIBS="$LIBS $PTHREAD_LIBS"
+        CC="$PTHREAD_CC"
+        AC_DEFINE(CONFIG_PTHREAD, 1, [Have POSIX threads])
+        pthread_found=true
+    ], [
+        AS_IF([test "x$enable_pthreads" = xyes], [
+            AC_MSG_ERROR([pthread support was requested, but was not found.])
+        ])
+    ])
+])
+
+AS_IF([test "x$pthread_found" != xtrue && test "x$enable_w32threads" != xno && test "x$ac_cv_header_stdatomic_h" = xyes], [
+    AS_IF([test "x$ac_cv_header_windows_h" = xyes], [
+        AC_DEFINE(CONFIG_W32THREADS, 1, [Have Win32 threads])
```

Comment:
pinging this since it prevents any Windows threading tests actually running in CI

Reply:
Fixed, but this *didn't* actually prevent the CI from testing threading on Windows, because the Windows autoconf builds all find and use pthreads, which are used preferentially over w32threads if available.

Reply:
right; but doesn't this mean nothing is currently testing the Win32 thread API paths?

Reply:
yes; not sure what the best solution there is, maybe I should `--disable-pthreads` on the UCRT64 build?

Reply:
> maybe I should --disable-pthreads on the UCRT64 build?
at least until/if this is supported with meson and it can be tested there with MSVC, probably yeah

Reply:
Yeah, that (or any other way to disable pthreads, if there are any other ways) sounds good to have in one of the test jobs.